src/main/java/org/folio/innreach/domain/service/impl/InnReachTransactionActionServiceImpl.java

```diff
@@ -123,6 +128,26 @@ public void linkNewLoanToOpenTransaction(LoanDTO loan) {
       });
   }
 
+  @Override
+  public void linkRenewalLoanToUpdateTransaction(LoanDTO loan) {
+    if (loan.getAction().equals("renewed")) {
+      var loanId = loan.getId();
+
+      transactionRepository.fetchAssociatedLoanWithInnReachTransaction(loanId)
+        .ifPresent(transaction -> {
+          log.info("Loan {} associated with transaction ", loanId);
+          var transactionHoldDueDate = Date.from(Instant.ofEpochSecond(transaction.getHold().getDueDateTime()));
+          int result = loan.getDueDate().compareTo(transactionHoldDueDate);
+          if (result != 0) {
+            var loanIntegerDueDate = (int) (loan.getDueDate().getTime()/1000);
```

Comment:
this check will not work as expected since loan's doe date includes milliseconds, you should truncate the value
```
          var transactionDueDate = Instant.ofEpochSecond(transaction.getHold().getDueDateTime());
          var loanDueDate = loan.getDueDate().toInstant().truncatedTo(ChronoUnit.SECONDS);

          if (!loanDueDate.equals(transactionDueDate)) {
```

Reply:
Done